cupy/core/core.pyx

```diff
@@ -36,6 +36,17 @@ cdef inline _should_use_rop(x, y):
     return xp < yp and not isinstance(y, ndarray)
 
 
+class IndexOrValueError(IndexError, ValueError):
+
+    def __init__(self, *args, **kwargs):
+        super(IndexOrValueError, self).__init__(*args, **kwargs)
+
+try:
+    AxisError = numpy.AxisError
+except AttributeError:
+    AxisError = IndexOrValueError
```

Comment:
`cupy.core.core.AxisError` should not be used directly, so an underscore should be added.